src/lib/utils/locking_allocator/locking_allocator.h

```diff
@@ -36,10 +36,20 @@ class BOTAN_PUBLIC_API(2,0) mlock_allocator final
 
       ~mlock_allocator();
 
+      friend struct mlock_allocator_initializer;
+
       std::unique_ptr<Memory_Pool> m_pool;
       std::vector<void*> m_locked_pages;
    };
 
+extern mlock_allocator& mlock_allocator_instance;
+
+// this class needs to remain in a public header, even if locking_allocator.h becomes internal
```

Comment:
Currently locking_allocator.h is public but it is only ever included in one source file (since we now abstract the allocation behind allocate_memory fn in `mem_ops.cpp`) and so without explicitly including it no application ever sees it. So is this still going to be effective? This may need to be in `secmem.h` instead.

Reply:
That's right. I assumed locking_allocator.h would be included indirectly but I haven't checked it.
After moving `mlock_allocator_initializer` to `secmem.h` I had problems with undefined references when building botan. It took me some time to figure out, why `mlock_allocator_initializer` gets added to the static library but not to the shared library. I realized that the declaration needs to be tagged with  `BOTAN_DLL` or one if its aliases.
Since I'm not sure which one to choose, I selected `BOTAN_DLL.` Please feel free to replace it with the appropriate one.